bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/item/ItemResource.java

```diff
@@ -367,6 +368,90 @@ public Response removeTag(@PathParam("itemname") String itemname, @PathParam("ta
         return Response.ok().build();
     }
 
+    /**
+     * Create or Update an item by supplying an item bean.
+     * @param itemname
+     * @param item the item bean.
+     * @return
+     */
+    @PUT
+    @Path("/{itemname: [a-zA-Z_0-9]*}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response createOrUpdateItem(@PathParam("itemname") String itemname, ItemBean item) {
+
+    	// If we didn't get an item bean, then return!
+    	if (item == null) {
+            return Response.status(Status.BAD_REQUEST).build();
+    	}
+
+        GenericItem newItem = null;
+
+        if (item.type != null && item.type.equals("GroupItem")) {
+            newItem = new GroupItem(itemname);
+        } else {
+        	String itemType = item.type.substring(0, item.type.length() - 4);
+            for (ItemFactory itemFactory : itemFactories) {
+                newItem = itemFactory.createItem(itemType, itemname);
+                if (newItem != null) {
+                    break;
+                }
+            }
+        }
+
+        if (newItem == null) {
+            logger.warn("Received HTTP PUT request at '{}' with an invalid item type '{}'.", uriInfo.getPath(),
+            		item.type);
+            return Response.status(Status.BAD_REQUEST).build();
+        }
+
+        // See if an existing item of this name exists.
+        Item existingItem = getItem(itemname);
+
+        // Update the label
+        if (item.label != null && !item.label.isEmpty()) {
+        	newItem.setLabel(item.label);
+        }
+        else if (existingItem != null) {
+        	newItem.setLabel(existingItem.getLabel());
```

Comment:
This would mean, that it is not possible to change the label to null of an existing item.

Also for the following lines, just do

```
newItem.setLabel(item.label);
newItem.setCategory(item.category);
...
```